frontend/src/auth/authenticator.ts

```diff
@@ -0,0 +1,19 @@
+export class Authenticator {
+  // This is a temporary variable that should functionally be replaced with a token that is stored in session storage, retrieved from the backend
+  static hasLoggedIn = false
+
+  static async isAuthenticated() {
+    // Should be replaced with getting something from session storage
+    return new Promise((resolve) => {
+      setTimeout(() => resolve(Authenticator.hasLoggedIn), 1000)
+    })
+  }
+
+  static async authenticate(username: string, password: string) {
+    // Should be replaced with a fetch request to backend
+    Authenticator.hasLoggedIn = true
+    return new Promise((resolve) => {
+      setTimeout(() => resolve(Authenticator.hasLoggedIn), 1000)
+    })
+  }
+}
```

Comment:
Maybe this should be a singleton instead of static methods and fields? Anyway, I would prefer using `useContext` hook or redux instead. [`useContext` hook](https://react.dev/reference/react/useContext) should be easier to setup.

Reply:
Changed to using useContext hook and performing authentication within home view

Reply:
Implemented auth functionality directly in auth context properties